radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/MintAndTransferTokensActionMapper.java

```diff
@@ -60,10 +89,31 @@ public MintAndTransferTokensActionMapper(
 			.map(TokenDefinitionsState.class::cast)
 			.map(TokenDefinitionsState::getState)
 			.map(state -> getTokenStateOrError(state, tokenDefinition))
-			.map(TokenState::getGranularity)
-			.map(TokenUnitConversions::unitsToSubunits)
-			.map(granularity -> createMint(mintTransferAction.getAmount(), granularity, tokenDefinition))
-			.map(mint -> mintAndTransferToGroupMapper.apply(mint, createTransfer(mint, mintTransferAction)))
+			.map(state -> {
+				final FungibleParticleTransition<UnallocatedTokensParticle, TransferredTokensParticle> mintTransition =
+					createMint(mintTransferAction.getAmount(), tokenDefinition, state);
+
+				final TransferredTokensParticle transferredTokensParticle = createTransfer(
+					state.getTokenSupplyType() == TokenSupplyType.FIXED
+						? ImmutableMap.of(
+							TokenTransition.MINT, TokenPermission.TOKEN_CREATION_ONLY,
```

Comment:
Same comment about name of transition applies here, I think it should be 'minting' and 'burning'

Reply:
disagree, we are moving to a system where transitions are becoming first class citizens in which case they should be named by their imperative, strongest version of the word